Feature event logging

zmelnick opened the pull request:
This is an open pull request for Event Logging. As tickets are closed and parts of Event Logging are implemented, this pull request will update. All people adding issues to this request please comment clearly on what you are doing,

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into CONNECT_integration

zmelnick committed:
Add Event interface for Event Logging. GATEWAY-2797

zmelnick committed:
Adding EventLogger interface for Event Logging. GATEWAY-2790

zmelnick committed:
Updating EventLogger to be abstract. GATEWAY-2790

zmelnick committed:
adding methods and implementions for EventLogger. GATEWAY-2790

zmelnick committed:
Adding EventManager for EventLogging. GATEAY-2791

zmelnick committed:
Making EventManager a singleton. GATEWAY-2791

zmelnick committed:
Updating singleton to be thread safe and load lazy. GATEWAY-2791

zmelnick committed:
Adding Log4jEventLogger, a concrete implementation of EventLogger for logging to log4j. GATEWAY-2794. Also includes a spelling correction on the Event class.

zmelnick committed:
Correcting EventLogger to properly represent correct types.

zmelnick committed:
Adding transactionID() to Event interface per request. Updaing classes as necessary. GATEWAY-2789

zmelnick committed:
Turning Log4jEventLogger into a singleton. Only one instance of this class should exist; better design to enforce that explicitly.

zmelnick committed:
Creating EventListenerFactory. Class currently only creates a Log4jEventLogger, as the outside logic supporting which class to instantiate hasn't been developed yet. GATEWAY-2795

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into feature-EventLogging

zmelnick committed:
Create appropriate Event Type implementaions for all lifecycle events. GATEWAY-2797

zmelnick committed:
changing Request/Response paradigm to Initiator/Responder. GATEWAY-2788

zmelnick committed:
Implementation of EventFactory. GATEWAY-2792

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into feature-EventLogging

zmelnick committed:
Changing package names for Events. GATEWAY-2788

zmelnick committed:
Removing dead code.

zmelnick committed:
Updating Events to remove transactionID, as that is retrieved dynamically and will not be passed in. GATEAY-2797

zmelnick committed:
Updating EventFactory to include updated understanding of TransactionID. Also formatting updates.

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
Revert "Removing dead code."

This reverts commit a5774160558fbdea1e4f507622723e478884b9c0.

zmelnick committed:
Revert "Updating Events to remove transactionID, as that is retrieved dynamically and will not be passed in. GATEAY-2797"

This reverts commit 68ca85b190f2848466c7cfb2b2163152b0f97890.

zmelnick committed:
Revert "Updating EventFactory to include updated understanding of TransactionID. Also formatting updates."

This reverts commit 9910710a6b7d3886aa5fd3d8b19fe64c9cae0ef3.

zmelnick committed:
Revert "Revert "Removing dead code.""

This reverts commit 000cd72a3d72c42724f8a0b186f231dd64bef3cd.

Commit:
Merge branch 'feature-EventLogging' of https://github.com/zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
Adding dependencies for aspectj and the application context for EventLogging

zmelnick committed:
merge conflicts

zmelnick committed:
updating .gitignore

Commit:
Merge branch 'feature-EventLogging' of https://github.com/zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
Renaming application context, moving aspectj jars around, and adding aspectj jars to be included in the ear.

zmelnick committed:
touch point for GATEWAY-2798

zmelnick committed:
resolving merge conflicts

Commit:
Merge branch 'feature-EventLogging' of https://github.com/zmelnick/CONNECT into feature-EventLogging

mlandis5am committed:
GATEWAY-2796: First version compiling, need to write some unit tests.

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into mvn-integration

Commit:
Added c3p0 jar in ear build script

mlandis5am committed:
GATEWAY-2796: Added unit test and verified installer creates/drops

zmelnick committed:
Merge pull request #1 from mlandis5am/GATEWAY-2796

Gateway 2796

zmelnick committed:
Updating to maven build

zmelnick committed:
touch point. working Aspect.

zmelnick committed:
touch point. working Aspect.

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
merge conflicts

mlandis5am committed:
GATEWAY-2796: Review changes.

zmelnick committed:
Merge pull request #2 from mlandis5am/GATEWAY-2796

GATEWAY-2796: Modify singleton to load lazy and be thread safe.

Commit:
Refactored EventFactory to be configurable via Spring Proxy

Commit:
Added Error Event.

zmelnick committed:
Merge pull request #3 from akong/feature-EventLogging

Refactored EventFactory to be configurable via Spring Proxy

zmelnick committed:
Merge pull request #3 from akong/feature-EventLogging

Refactored EventFactory to be configurable via Spring Proxy

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

mlandis5am committed:
GATEWAY-2795: First commit for event logger factory

zmelnick committed:
updating .gitignore

mlandis5am committed:
GATEWAY-2795: Added bootstrapping, configuration, refactored EventFactory for consistency

mlandis5am committed:
GATEWAY-2795: Refactored the component proxy factory a little, more bootstrapping, verified with debugger

Commit:
Added join points and msg id retrieval to EventAspect.  Added aspect to DocSubmissionProxyConfig.xml.

zmelnick committed:
cleanup code

zmelnick committed:
merge conflicts

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

Conflicts:
	Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspect.java
	Product/Production/Gateway/DocumentSubmission_20/src/main/webapp/WEB-INF/cxf-servlet.xml

mlandis5am committed:
GATEWAY-2795: Merging in complete code from feature-Eventlogging, including mvn

Commit:
Merge pull request #4 from mlandis5am/GATEWAY-2795

Gateway 2795

Commit:
Created service specific aspects.  Added message id and transaction id retrieval logic.

zmelnick committed:
merge conflicts
:

zmelnick committed:
Merge pull request #5 from akong/feature-EventLogging

Added join points and msg id retrieval to EventAspect.  Added aspect to ...

zmelnick committed:
merge conflits

zmelnick committed:
updating wsdl classpath on cxf-servelet

Commit:
Refactored and added outbound to EventAspect

zmelnick committed:
Merge pull request #6 from akong/feature-EventLogging

Refactored and added outbound to EventAspect

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
cleanup code

zmelnick committed:
refactoring transactionLogging instance to updated method name

zmelnick committed:
refactoring of Event Aspect and re-organization of DocSubmissionEventAspect

zmelnick committed:
updaing cxf servlet.

Commit:
Reverted changes to Nhin DS

zmelnick committed:
Merge pull request #7 from akong/feature-EventLogging

Reverted changes to Nhin DS

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
merge conflicts

Commit:
fixed cxf-servlet.xml

Commit:
Merge pull request #8 from akong/feature-EventLogging

fixed cxf-servlet.xml

zmelnick committed:
changes to cxf-servlet

zmelnick committed:
merge conflicts

Commit:
Implemented business logic for Event Aspect.  Added response id and action to event description.

Commit:
Merge pull request #9 from akong/feature-EventLogging

Implemented business logic for Event Aspect.  Added response id and acti...

Commit:
GATEWAY-2847: Added event triggering for DQ

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

Commit:
GATEWAY-2848: Implemented aspects for PD (no deferred yet)

Commit:
Merge pull request #10 from akong/feature-EventLogging

GATEWAY-2847: Added event triggering for DQ

Commit:
GATEWAY-2848: Added event logging to PD Deferred Req/Resp

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

zmelnick committed:
Merge pull request #11 from akong/feature-EventLogging

GATEWAY-2848: Added event logging to PD Deferred Req/Resp

zmelnick committed:
Create event triggering for DS. GATEWAY-2845

zmelnick committed:
merge conflicts

zmelnick committed:
cleanup, and fixing bootstrapping

Commit:
GATEWAY-2846: Added event logging to DR.  Fixed database issue with events.

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

bhumphrey commented on Product/Production/Adapters/AdminDistribution_a0/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,27 @@
```
shouldn't be committed

bhumphrey commented on Product/Production/Adapters/DocumentQuery_a0/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,30 @@
+Manifest-Version: 1.0
```
this is generated by the build. this shouldn't be committed.

bhumphrey commented on Product/Production/Adapters/DocumentRetrieve_a0/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,30 @@
+Manifest-Version: 1.0
```
same.

bhumphrey commented on Product/Production/Adapters/DocumentSubmission_a0/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,27 @@
+Manifest-Version: 1.0
```
same

bhumphrey commented on Product/Production/Adapters/General/CONNECTAdapterWeb/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,32 @@
+Manifest-Version: 1.0
```
same

bhumphrey commented on Product/Production/Adapters/General/CONNECTConsumerPreferencesProfileGUI/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,44 @@
+Manifest-Version: 1.0
```
same

bhumphrey commented on Product/Production/Adapters/General/CONNECTDeferredQueueManagerGUI/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,43 @@
+Manifest-Version: 1.0
```
same

bhumphrey commented on Product/Production/Adapters/General/CONNECTUniversalClientGUI/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,47 @@
+Manifest-Version: 1.0
```
same

bhumphrey commented on Product/Production/Adapters/PatientDiscovery_a0/src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,28 @@
+Manifest-Version: 1.0
```
same

Commit:
GATEWAY-2848: Fixed PD unit tests

Commit:
Merge pull request #12 from zmelnick/feature-EventLogging_working

Create event triggering for DS

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

zmelnick committed:
merge conflicts

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
Merge pull request #13 from akong/feature-EventLogging

Added event logging to DR.  Fixed event database sql script.

zmelnick committed:
merge conflicts

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into feature-EventLogging

zmelnick committed:
Create triggers for AD. GATEWAY-2912

zmelnick committed:
GATEWAY-2912

zmelnick committed:
refactor HIEM package structure to match HIEM_Core (service first view). Also includes work for HIEM Unsubscribe EventLogging (working, but incomplete)

zmelnick committed:
bug fix for AD Event Aspect. Removing proxy call from nwhin invocation, and changing a misnamed package from entity -> nhin

zmelnick committed:
Event Logging Triggers for Unsubscribe. GATEWAY-2913

zmelnick committed:
GATEWAY-2913

zmelnick committed:
Merge pull request #14 from zmelnick/feature-EventLogging_working

Feature event logging working

rhalfertycgi committed:
Adding Event Logging for Subscribe GATEWAY-2914

rhalfertycgi committed:
Adding Event Logging for Subscribe GATEWAY-2914

Conflicts:
	Product/Production/Gateway/HIEM_20/src/main/webapp/WEB-INF/cxf-servlet.xml
	Product/Production/Services/HIEMCore/src/main/java/gov/hhs/fha/nhinc/subscribe/aspect/SubscribeEventAspect.java

zmelnick committed:
Touch point for GATEWAY-2915

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
Merge branch 'feature-EventLogging_working' into feature-EventLogging

zmelnick committed:
merge conflicts with CONNECT_integration

zmelnick committed:
removing ear file from previous merge.

rhalfertycgi commented on Product/Production/Gateway/DocumentSubmission_20/src/main/webapp/WEB-INF/cxf-servlet.xml:
```diff
@@ -38,25 +46,39 @@ http://cxf.apache.org/jaxws http://cxf.apache.org/schemas/jaxws.xsd">
 
     <!-- Entity services -->
+
+    <bean id="entityDocSubmissionOrchImpl" class="gov.hhs.fha.nhinc.docsubmission.entity.EntityDocSubmissionOrchImpl" />
```
The g0 and g1 services seem to be implemented differently.

mlandis5am committed:
GATEWAY-2916: Refactored single test to verify aspect pointcuts

mlandis5am committed:
Merge pull request #16 from mlandis5am/GATEWAY-2916

GATEWAY-2916: Refactored single test to verify aspect pointcuts

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
fixing bad merge conflict

rhalfertycgi committed:
Fixing bugs in Event Logging GATEWAY-2914

rhalfertycgi committed:
Fixing bugs in Event Logging GATEWAY-2914

Conflicts:
	Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/AdminDistributionEventAspect.java
	Product/Production/Services/HIEMCore/src/main/java/gov/hhs/fha/nhinc/unsubscribe/aspect/UnsubscribeEventAspect.java

rhalfertycgi committed:
merge conflicts

rhalfertycgi committed:
Resolving merge conflicts with feature branch

zmelnick committed:
cleanup

zmelnick committed:
merge conflicts

rhalfertycgi committed:
Fixing code comments in Event Logging GATEWAY-2914

rhalfertycgi committed:
Fixing code comments in Event Logging GATEWAY-2914

zmelnick committed:
Merge pull request #17 from rhalfertycgi/CONNECT_integration

Adding event logging and fixes for bugs

zmelnick committed:
ensuring that UnsubscribeEventAspect works for all orchestration interfaces.

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

Commit:
merge conflicts

Commit:
fixed build and deploy errors

Commit:
merge conflicts

Commit:
merged changes

Commit:
Merge pull request #18 from bhumphrey/feature-EventLogging

fixed build and deploy errors

zmelnick committed:
cleanup

zmelnick committed:
cleanup

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

zmelnick committed:
GATEWAY-2935

zmelnick committed:
GATEWAY-2935

zmelnick committed:
Merge branch 'CONNECT_integration' of https://github.com/CONNECT-Solution/CONNECT into feature-EventLogging

Commit:
adding building around event description

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

zmelnick committed:
adding NotifyEventAspect, and cleaning up merge conflicts

zmelnick committed:
touch point. saving work

zmelnick committed:
merge conflicts.

zmelnick committed:
merge conflicts

zmelnick committed:
Merge branch 'savepoint' into feature-EventLogging

zmelnick committed:
Revert "Merge branch 'savepoint' into feature-EventLogging"

This reverts commit 2bfe3c4195c9ef62c88b0c0695320b04b5a2ef14, reversing
changes made to aead4c567949a5082a7e7fddf4fe4b3dad945f96.

zmelnick committed:
save point

zmelnick committed:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into working

zmelnick committed:
update

zmelnick committed:
update

zmelnick committed:
update

zmelnick committed:
update

zmelnick committed:
refactoring SubscribeEventAspect to use new name for EventAspect

zmelnick committed:
fixing merge conflicts

zmelnick committed:
merge conflicts

Commit:
GATEWAY-2903: added description and event builder pattern, modified how event factory works

Commit:
changed eclipse plugin to use project dependencies

Commit:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

Conflicts:
	Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java
	Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/event/EventLoggerFactory.java
	Product/Production/Gateway/PatientDiscovery_10/pom.xml
	Product/Production/Gateway/PatientDiscovery_10/src/main/webapp/META-INF/MANIFEST.MF
	Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/utils/PDTestUtils.java

Commit:
Merge remote-tracking branch 'zmelnick/feature-EventLogging' into feature-EventLogging

Commit:
Build fix for merge issues by adding EntityDocRetrieveOrchImpl.  May still have problems during runtime

zmelnick committed:
Merge pull request #20 from akong/feature-EventLogging

Feature event logging

Commit:
Merge pull request #21 from bhumphrey/feature-EventLogging

correct compilation errors

Commit:
adding missing file

Commit:
GATEWAY-2903: change dependency inject for PDEvents

Commit:
Merge branch 'feature-EventLogging' of github.com:zmelnick/CONNECT into feature-EventLogging

Conflicts:
	Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/EntityDocRetrieveOrchImpl.java

zmelnick committed:
Merge pull request #22 from bhumphrey/feature-EventLogging

added missing file after move merge confusion

bhumphrey commented:
closing to merge to feature-EventLogging
